ethereum-contracts/contracts/ChaosnetChallenges/Puzzle.sol

```diff
@@ -0,0 +1,109 @@
+pragma solidity ^0.5.12;
+
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "darknode-sol/contracts/Shifter/ShifterRegistry.sol";
+import "darknode-sol/contracts/libraries/Compare.sol";
+
+contract Puzzle is Ownable {
+    using SafeMath for uint256;
+
+    bytes public secretHash;
+    uint256 public rewardAmount;
+    string public tokenSymbol;
+    bool public rewardClaimed;
+
+    ShifterRegistry public registry;
+
+    /// @param _registry The Shifter registry contract address.
+    /// @param _tokenSymbol The shifter token symbol
+    /// @param _secretHash The secret hash
+    constructor(
+        ShifterRegistry _registry,
+        string memory _tokenSymbol,
+        bytes memory _secretHash
+    ) public {
+        registry = _registry;
+        tokenSymbol = _tokenSymbol;
+        secretHash = _secretHash;
+    }
+
+    /// @notice Allows someone to try and claim the reward by submitting the secret.
+    /// @param _amount The amount of Bitcoin provided to the Darknodes in Sats.
+    /// @param _nHash The hash of the nonce returned by the Darknodes.
+    /// @param _sig The signature returned by the Darknodes.
+    function fund(
+        // Required
+        uint256      _amount,
+        bytes32      _nHash,
+        bytes memory _sig
+    ) public {
+        require(_amount > 0, "amount must be greater than 0");
+        uint256 transferAmount = registry.getShifterBySymbol(tokenSymbol).shiftIn(0x0, _amount, _nHash, _sig);
+        rewardAmount = rewardAmount.add(transferAmount);
+    }
+
+    /// @notice Allows someone to try and claim the reward by submitting the secret.
+    /// @param _refundAddress The address that should receive the shiftedOut tokens and the potential reward.
+    /// @param _secret The secret.
+    /// @param _amount The amount of Bitcoin provided to the Darknodes in Sats.
+    /// @param _nHash The hash of the nonce returned by the Darknodes.
+    /// @param _sig The signature returned by the Darknodes.
+    function claimReward(
+        // Payload
+        bytes memory _refundAddress,
```

Comment:
Using `_refundAddress` as the name is probably misleading. This is the address that will receive the claimed reward (there’s no “refund” happening).

Reply:
But if the secret is incorrect then the funds that they shifted in is "refunded" back to them, no? Do you think `_rewardAddress` or something is more appropriate?